nova/tests/unit/virt/vmwareapi/test_vmops.py

```diff
@@ -4076,3 +4081,67 @@ def specced_flavor(specs):
         extra_specs = self._vmops._get_extra_specs(
             specced_flavor({'trait:CUSTOM_NUMASIZE_C48_M729': 'forbidden'}))
         self.assertEqual(extra_specs.numa_prefer_ht, '')
+
+    def _test_stack_vm_to_host_if_needed(self, instance_memory_mb,
+                                         hosts, expected):
+        with test.nested(
+                mock.patch.object(vm_util,
+                                  'get_stats_from_cluster_per_host',
+                                  return_value=hosts),
+                mock.patch.object(cluster_util, 'is_drs_enabled',
+                                  return_value=False),
+        ) as (_get_stats_from_cluster_per_host, _is_drs_enabled):
+            instance = self._instance.obj_clone()
+            instance.memory_mb = instance_memory_mb
+
+            host_ref = self._vmops._stack_vm_to_host_if_needed(instance)
+
+            _get_stats_from_cluster_per_host.assert_called_once_with(
+                self._session, self._cluster.obj)
+            _is_drs_enabled.assert_called_once_with(self._session,
+                                                    self._cluster.obj)
+            self.assertEqual(expected, host_ref.value if host_ref else None)
+
+    def test_stack_vm_to_host_if_needed(self):
+        hosts = {
+            'host-1': {
+                'name': 'host2',
+                'memory_mb': 4096,
+                'memory_mb_used': 1024,
+                'memory_mb_reserved': 512,
+            },
+            'host-2': {
+                'name': 'host2',
+                'memory_mb': 4096,
+                'memory_mb_used': 1024,
+                'memory_mb_reserved': 1024,
+            }
+        }
+
+        self._test_stack_vm_to_host_if_needed(
+            1024, hosts, 'host-2')
+        self._test_stack_vm_to_host_if_needed(
+            2048, hosts, 'host-2')
+        self._test_stack_vm_to_host_if_needed(
+            2560, hosts, 'host-1')
+        self._test_stack_vm_to_host_if_needed(
+            3072, hosts, None)
+
+        hosts = {
+            'host-1': {
+                'name': 'host1',
+                'memory_mb': 4096,
+                'memory_mb_used': 1024,
+                'memory_mb_reserved': 1024,
+            },
+            'host-2': {
+                'name': 'host2',
+                'memory_mb': 4096,
+                'memory_mb_used': 1024,
+                'memory_mb_reserved': 1024,
+            }
+        }
+        self._test_stack_vm_to_host_if_needed(
+            1024, hosts, 'host-1')
+        self._test_stack_vm_to_host_if_needed(
+            2048, hosts, 'host-1')
```

Comment:
Are you testing here, that the first host gets returned if both have the same fill-grade? I would have thought we just need "not None" because in the end we do not really care, no?

In any, case, it would be good if you could add a comment that states what you're testing here.